usecases/auth/authorization/rbac/manager.go

```diff
@@ -281,3 +313,20 @@ func prettyStatus(value bool) string {
 	}
 	return "failed"
 }
+
+func collectStaleRoles(polices [][]string, casbinStoragePoliciesMap map[string]struct{}, casbinStoragePolicies [][][]string) [][][]string {
+	for _, p := range polices {
+		// ignore builtin roles
+		if slices.Contains(authorization.BuiltInRoles, conv.TrimRoleNamePrefix(p[1])) {
+			continue
+		}
+		// collect stale or empty roles
+		if _, ok := casbinStoragePoliciesMap[p[1]]; !ok {
+			// e.g. policy line in casbin -> g, user:wv_internal_empty, role:roleName, that's why p[1]
+			casbinStoragePolicies = append(casbinStoragePolicies, [][]string{{
+				p[1], conv.InternalPlaceHolder, conv.InternalPlaceHolder, "*",
+			}})
+		}
+	}
+	return casbinStoragePolicies
+}
```

Comment:
**Question (non-blocking)**: Likewise above, this will loop over every policy to find those that are assigned to `conv.InternalPlaceHolder`. Any potential for performance penalty here, do you think?

Reply:
those are the policy returned from casbin, so in case `getAllRoles()` it will be all policies yes and in case of `getRoleByName()` it will get the role and the grouping of this role only